### PR TITLE
feat: hands feature fixes

### PR DESCRIPTION
**Pull request by Kinerius**
## What does this PR change?

Old upper-bodies will hide the base hands by default.

Updated hands-wear icon

Fixed category naming

## How to test the changes?

Enter https://play.decentraland.zone/?NETWORK=goerli&realm=artemis&explorer-branch=feat/hands-default-hide
- Check out the hands section in the backpack.
- Equip and unequip them correctly.
- Check that all names are `HANDWEAR`
- Check that wearables that hide hands have the hands icon, not the gloves icon, those are for handwear

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 726bd0f</samp>

This pull request adds a new feature to the wearables catalog service and the `WearableItem` class to allow wearables to override the default hiding behavior of some categories. This enables more avatar customization options and improves the appearance of some wearables.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs [hunk 1]
```diff
@@ -231,6 +238,14 @@ public string[] GetHidesList(string bodyShapeType)
                 : hides.Concat(SKIN_IMPLICIT_CATEGORIES).Distinct().ToArray();
         }
 
+        // Old UPPER_BODY Wearables are incompatible with the new Hands category,
+        // so we apply this rule to hide the hands by default
+        bool isOrHidesUpperBody = hides.Contains(Categories.UPPER_BODY) || data.category == Categories.UPPER_BODY;
+        bool removesDefault = data.removesDefaultHiding?.Contains(Categories.HANDS) ?? false;
+
+        if (isOrHidesUpperBody && !removesDefault)
+            hides = hides.Concat(UPPER_BODY_DEFAULT_HIDES).Distinct().ToArray();
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs:247]: we might be able to improve GC if we store the hides information in a pre-allocated buffer

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs [hunk 2]
```diff
@@ -231,6 +238,14 @@ public string[] GetHidesList(string bodyShapeType)
                 : hides.Concat(SKIN_IMPLICIT_CATEGORIES).Distinct().ToArray();
         }
 
+        // Old UPPER_BODY Wearables are incompatible with the new Hands category,
+        // so we apply this rule to hide the hands by default
+        bool isOrHidesUpperBody = hides.Contains(Categories.UPPER_BODY) || data.category == Categories.UPPER_BODY;
+        bool removesDefault = data.removesDefaultHiding?.Contains(Categories.HANDS) ?? false;
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs:244]: its hard to understand why the naming of `removesDefault`

Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs:244]: Yeah the concept of this `removesDefaultHiding` is super corner case-y, it's the workaround we found for hiding the base hands by default and then removing that default if the upper body is a newer version.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs (hunk 1 above)
Comment (Kinerius) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs]: I didn't want to touch the original implementation but now that you mention it, all of this should be just a `hashset`

**Review by AjimenezDCL — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs [hunk 3]
```diff
@@ -52,7 +52,7 @@ public class WearableItem
         { Categories.EYEBROWS, "Eyebrows" },
         { Categories.BODY_SHAPE, "Body shape" },
         { Categories.FACIAL_HAIR, "Facial hair" },
-        { Categories.HANDS_WEAR, "Hands" },
+        { Categories.HANDS_WEAR, "Handwear" },
```
Comment (davidejensen) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Models/AvatarAssets/WearableItem.cs:55]: Handwear singular? Not Handswear plural?

**Review by davidejensen — APPROVED** [review]

**Pull request merged**